Cpp/Patterns/src/MatrixConnectedRowsAndColumns.lf

```diff
@@ -1,75 +1,77 @@
+// hierachy to implement the second dimension. Nodes are organized in Rows which are grouped to form
+// the matrix.
+target Cpp
 
 public preamble {=
-    struct Pos {
-        size_t col;
-        size_t row;
-    };
+  struct Pos {
+      size_t col;
+      size_t row;
+  };
 
-    std::ostream& operator<<(std::ostream& os, const Pos& pos);
+  std::ostream& operator<<(std::ostream& os, const Pos& pos);
 =}
 
 private preamble {=
-    std::ostream& operator<<(std::ostream& os, const Pos& pos) {
-        os << '(' << pos.col << ',' << pos.row << ')';
-        return os;
-    }
+  std::ostream& operator<<(std::ostream& os, const Pos& pos) {
+      os << '(' << pos.col << ',' << pos.row << ')';
+      return os;
+  }
 =}
 
-reactor Node(bank_index: size_t(0), row_index: size_t(0), num_rows: size_t(4), num_cols:size_t(4)) {
-    input[num_cols] fromRow: Pos
-    input[num_rows] fromCol: Pos
+reactor Node(
+  bank_index: size_t(0),
+  row_index: size_t(0),
+  num_rows: size_t(4),
+  num_cols: size_t(4)
+) {
+  input[num_cols] fromRow: Pos
+  input[num_rows] fromCol: Pos
 
-    output toRowAndCol: Pos
+  output toRowAndCol: Pos
 
-    state pos: Pos{bank_index, row_index}
+  state pos: Pos{bank_index, row_index}
 
-    reaction (startup) -> toRowAndCol {=
-        std::cout << "Hello from " << pos << '\n';
-        // send my position to everyone else in my row and column
-        toRowAndCol.set(pos);
-    =}
+  reaction(startup) -> toRowAndCol {=
+    std::cout << "Hello from " << pos << '\n';
+    // send my position to everyone else in my row and column
+    toRowAndCol.set(pos);
+  =}
 
-    reaction (fromRow) {=
-        std::cout << pos << " received row messages from: ";
-        for (auto& port : fromRow) {
-            if (port.is_present()) {
-                std::cout << *port.get() << ", ";
-            }
+  reaction(fromRow) {=
+    std::cout << pos << " received row messages from: ";
+    for (auto& port : fromRow) {
+        if (port.is_present()) {
+            std::cout << *port.get() << ", ";
         }
-        std::cout << '\n';
-    =}
+    }
+    std::cout << '\n';
+  =}
 
-    reaction (fromCol) {=
-        std::cout << pos << " received col messages from: ";
-        for (auto& port : fromCol) {
-            if (port.is_present()) {
-                std::cout << *port.get() << ", ";
-            }
+  reaction(fromCol) {=
+    std::cout << pos << " received col messages from: ";
+    for (auto& port : fromCol) {
+        if (port.is_present()) {
+            std::cout << *port.get() << ", ";
         }
-        std::cout << '\n';
-    =}
+    }
+    std::cout << '\n';
+  =}
 }
 
-reactor Row(bank_index: size_t(0), num_rows:size_t(4), num_cols:size_t(4)) {
-    nodes = new[num_cols] Node(row_index=bank_index, num_rows=num_rows, num_cols=num_cols)
+reactor Row(bank_index: size_t(0), num_rows: size_t(4), num_cols: size_t(4)) {
+  nodes = new[num_cols] Node(row_index = bank_index, num_rows = num_rows, num_cols = num_cols)
```

Comment:
In Python typically the argument assignments are formatted without a space around the `=`. This is also part of the Google Python style. I don't know how other languages with a similar syntax handle this. We might want to consider removing the spaces, since we want to be close to the Google style, but I don't feel strongly about this.

Reply:
One consideration is that we surround the ` = ` with a space in the definition of the reactor, and we really should do it that way because there are also type annotations, which would be hard to read without the whitespace. We might choose to use the spaces where we instantiate the reactors as well, and in the Python target, in the name of consistency.

Reply:
The Google style guide says that for the definition there should be a space if there is a type annotation and no space if there is none.

I am fine with leaving this as is, but if we say we take the Google style guide as a reference, then maybe we should open an issue collecting this instances where we deviate from the Google style and fix those at some point when we feel this is important.